lib/Frontend/Frontend.cpp

```diff
@@ -258,8 +258,6 @@ ModuleDecl *CompilerInstance::getMainModule() {
 
     if (Invocation.getFrontendOptions().EnableResilience)
       MainModule->setResilienceStrategy(ResilienceStrategy::Resilient);
-    else if (Invocation.getSILOptions().SILSerializeAll)
-      MainModule->setResilienceStrategy(ResilienceStrategy::Fragile);
```

Comment:
You forgot to remove ResiienceStrategy::Fragile

Reply:
@slavapestov In cases like below, should we check instead for SILSerializeWitnessTables or remove them alltogether?  
```c++
if (isProtocolConformanceKind(getKind())) {
    auto conformance = getProtocolConformance();

    auto conformanceModule = conformance->getDeclContext()->getParentModule();
    auto isCompletelySerialized = conformanceModule->getResilienceStrategy() ==
                                  ResilienceStrategy::Fragile;

    // The conformance is fragile if it is in a -sil-serialize-all module.
    return isCompletelySerialized;
  }
```

Reply:
I think you should check your flag yes.

Reply:
@slavapestov Hmm. There is a small problem here. The code above is inside `LinkEntity::isFragile`, which does not contain any reference to the `SILModule`, which has `SILSerializeWitnessTables ` flag. It can only access `ModuleDecl` through `conformance->getDeclContext()->getParentModule()`.

How do we solve this in the cleanest way? Should we add an `IGM` argument to the function? Should we add a flag for -sil-serialize-witness-tables and -sil-serialize-vtables to the `ModuleDecl` or may be even have a reference from `ModuleDecl` to `SILOptions`?

BTW, why don't `ModuleDecls` have a back-reference to the `SILModules`? It could be useful for such cases...

Reply:
OK. I moved `SILSerializeWitnessTables` and `SILSerializeVTables` into `LanguageOptions`, so that they are accessible from `ASTContext` and `ModuleDecl`.